packages/netlify-cms-core/src/actions/editorialWorkflow.ts

```diff
@@ -386,7 +386,9 @@ export function persistUnpublishedEntry(collection: Collection, existingUnpublis
      * update the entry and entryDraft with the serialized values.
      */
     const fields = selectFields(collection, entry.get('slug'));
-    const serializedData = serializeValues(entry.get('data'), fields);
+    let serializedData = serializeValues(entry.get('data'), fields);
```

Comment:
We should apply this fix also here:
https://github.com/netlify/netlify-cms/blob/684b79e43bebb63ce1e844eae5c8c0e76087687b/packages/netlify-cms-core/src/actions/entries.ts#L617

Right?

Also, what do you think about moving this fix into the `serializeValues` and `deserializeValues` functions? 

Should be easier to add unit tests for that as well.